core/src/test/scala/integration/kafka/api/SaslSetup.scala

```diff
@@ -60,27 +65,39 @@ trait SaslSetup {
       kdc.start()
       kdc.createPrincipal(serverKeytabFile, JaasTestUtils.KafkaServerPrincipalUnqualifiedName + "/localhost")
       kdc.createPrincipal(clientKeytabFile, JaasTestUtils.KafkaClientPrincipalUnqualifiedName, JaasTestUtils.KafkaClientPrincipalUnqualifiedName2)
-    } else {
-      this.clientKeytabFile = None
-      this.serverKeytabFile = None
     }
-    setJaasConfiguration(mode, kafkaServerJaasEntryName, kafkaServerSaslMechanisms, kafkaClientSaslMechanism)
+    if (withDefaultJaasContext) {
+      createJaasConfiguration(mode, kafkaServerJaasEntryName, kafkaServerSaslMechanisms, kafkaClientSaslMechanism)
+      setJaasConfiguration()
+    } else
+        createJaasConfiguration(mode, kafkaServerJaasEntryName, kafkaServerSaslMechanisms, kafkaClientSaslMechanism)
     if (mode == Both || mode == ZkSasl)
       System.setProperty("zookeeper.authProvider.1", "org.apache.zookeeper.server.auth.SASLAuthenticationProvider")
   }
 
-  protected def setJaasConfiguration(mode: SaslSetupMode, kafkaServerEntryName: String,
+  protected def createJaasConfiguration(mode: SaslSetupMode, kafkaServerEntryName: String,
```

Comment:
I renamed it because it just creates a new JaasConfiguration but does not write it to the file. SaslSetup has a setJaasConfiguration which gets the JaasConfiguration created by the above mentoned method and writes it to the file.